packages/components/src/useScreenSize/getVariant.js

```diff
@@ -0,0 +1,19 @@
+import breakpoints from './breakpoints';
+
+const getVariant = () => {
+    const width = window.innerWidth;
+
+    let result;
+
+    Object.entries(breakpoints).some(([ variant, size ]) => {
+        if (width > size) {
+            result = variant;
+        } else {
+            return true;
+        }
+    });
+
+    return result;
```

Comment:
Why not use `find` instead of assign and somewhat complicated `some`?
```JS
return Object.entries(breakpoints).find(([ variant, size ]) => width > size)?.[0];
```

Reply:
That would find just the first variant that is bigger than the current size, we need to find the last one. We could reverse the  array first but I think that would make it somewhat complicated. This `some` will do the minimal required numbers of cycles without a need to reverse anything.

Reply:
Using `some` this way is basically a while cycle you can end when you need it.

Reply:
I had to run it locally to see what it actually does tbh. It's a clever way of how to use the `some` function, but I think it's a bit complicated to read. Because what this function does is finding the last breakpoint which is smaller than the current screen size. 
```JS
const breakpoints = {
  'sm': 100,
  'md': 200,
  'xl': 300,
  'xxl': 400
}

const getVariant = (width) => {
  let result;
​
    Object.entries(breakpoints).some(([ variant, size ]) => {
        if (width > size) {
            result = variant;
        } else {
            return true;
        }
    });
​
    return result;
}

getVariant(150); // returns 'sm'
getVariant(350); // returns 'xl'
```

I know we shouldn't use `break`, but this might be a bit more straight forward
```JS
const getVariant2 = (width) => {
  let result;
  for (const [variant, size] of Object.entries(breakpoints)) {
    if (size > width) {
      break;
    }
  result = variant;
}

getVariant2(150); // returns 'sm'
getVariant2(350); // returns 'xl'
```

Reply:
small correction: `if (size >= width) {`

Yeah, I wanted to use `break` initially but using some was much cooler. 😎